multi-node/aws/pkg/config/config.go

```diff
@@ -385,26 +398,32 @@ func (cfg Cluster) valid() error {
 		return fmt.Errorf("invalid vpcCIDR: %v", err)
 	}
 
-	_, instancesNet, err := net.ParseCIDR(cfg.InstanceCIDR)
-	if err != nil {
-		return fmt.Errorf("invalid instanceCIDR: %v", err)
-	}
-	if !vpcNet.Contains(instancesNet.IP) {
-		return fmt.Errorf("vpcCIDR (%s) does not contain instanceCIDR (%s)",
-			cfg.VPCCIDR,
-			cfg.InstanceCIDR,
-		)
-	}
-
 	controllerIPAddr := net.ParseIP(cfg.ControllerIP)
 	if controllerIPAddr == nil {
 		return fmt.Errorf("invalid controllerIP: %s", cfg.ControllerIP)
 	}
-	if !instancesNet.Contains(controllerIPAddr) {
-		return fmt.Errorf("instanceCIDR (%s) does not contain controllerIP (%s)",
-			cfg.InstanceCIDR,
-			cfg.ControllerIP,
-		)
+
+	for i, subnet := range cfg.Subnets {
+		if subnet.AvailabilityZone == "" {
+			return fmt.Errorf("availabilityZone must be set for subnet #%d", i)
+		}
+		_, instancesNet, err := net.ParseCIDR(subnet.InstanceCIDR)
+		if err != nil {
+			return fmt.Errorf("invalid instanceCIDR for subnet #%d: %v", i, err)
+		}
+		if !vpcNet.Contains(instancesNet.IP) {
+			return fmt.Errorf("vpcCIDR (%s) does not contain instanceCIDR (%s) for subnet #%d",
+				cfg.VPCCIDR,
+				cfg.InstanceCIDR,
+				i,
+			)
+		}
+		if i == 0 && !instancesNet.Contains(controllerIPAddr) {
```

Comment:
Thanks for your comment!
I agree that we are better documenting it for now.

However we should also remove it when we introduce an auto-scaling group to manage controller(s) and an ELB pointing to it.
This "convention" remains here because we currently have a singleton controller(which has `controllerIP` pointing to it) and it is placed in the first subnet "for now"(choosing one of subnets for the singleton controller doesn't make sense but I have proceeded with this convention for now).

If we had an auto-scaling group managing controller(s), we would configure the group to spread EC2 instances evenly over multiple subnets(as I had done for workers in this PR).
Then, we don't need to choose first/second/last subnet to place the singleton controller or to specify/validate controllerIP(because we don't need it anymore. We will point the master from workers via DNS name of an load balancer or Route53 DNS name pointing to it).

Does this make sense to you? @aaronlevy @colhom